hack/gen-logs.sh

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# ---------- config ----------
+# Paths (relative to repo root) to exclude.
+# Each entry excludes that directory AND all its subdirectories.
+EXCLUDED_PATH_PREFIXES=(
```

Comment:
If this could also exclude paths listed in gitignore, then it would be somewhat better positioned for future directories to ignore